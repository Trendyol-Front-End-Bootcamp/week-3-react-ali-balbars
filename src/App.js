import "./css/styles.css";
import React, { useEffect, useState, useRef } from "react";
import CharacterList from "./components/CharacterList/CharacterList.js";
import CharacterDetail from "./components/CharacterDetail/CharacterDetail.js";
import Filter from "./components/Filter/Filter.js";
import Header from "./components/Header/Header.js";
import NoResult from "./components/NoResult/NoResult.js";
import { BrowserRouter as Router, Link, Route } from "react-router-dom";

export default function App() {
  const [characters, setCharacters] = useState([]);
  const [filter, setFilter] = useState({
    search: '',
    gender: '',
    aliveness: '',
    species: '',
  });

  const generateUrl = () => {
    let url = `https://rickandmortyapi.com/api/character/?gender=${filter.gender ? filter.gender : ''}&species=${filter.species ? filter.species : ''}&status=${filter.aliveness ? filter.aliveness : ''}`;
    return url;
  }

  useEffect(() => {
    fetch("https://rickandmortyapi.com/api/character")
      .then((response) => {
        return response.json();
      })
      .then((response) => {
        setCharacters(response.results);
      });
  }, []);

  return (
    <Router>
      <Route
        path="/"
        exact
        render={() => {
          {
            return (
              <div className="App">
                <Header></Header>
                <div>
                  <Filter
                    filter={filter}
                    setFilter={setFilter}
                  />
                  <CharacterList
                    characters={characters}
                    filter={filter}
                  />
                </div>
              </div>
            );
          }
        }}
      />

      <Route
        path="/character/:name"
        render={(renderProps) => {
          const character = characters.find(
            (character) =>
              character.name.replace(/(\w+)\s(\w+)/, "$1-$2").toLowerCase() === renderProps.match.params.name
          );
          return (
            <CharacterDetail
              {...renderProps}
              character={character}
            />
          );
        }}
      />
      <Route
        path="/no-result"
        render={() => {
          return <NoResult filter={filter}
            setFilter={setFilter}
            generateUrl={generateUrl} />
        }}

      />
    </Router>
  );
}
