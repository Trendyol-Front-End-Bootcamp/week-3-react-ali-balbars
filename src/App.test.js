import App from './App';
import { render, screen, cleanup } from '@testing-library/react';
import React from 'react';
import { act } from 'react-dom/test-utils';

// describe('App Component', () => {
//     // afterEach(cleanup)
//     it('renders correctly', () => {
//         // act(() => {
//         //     render(<App></App>);
//         // })

//         // await act( async () => render(<App></App>))
//         // render(<App></App>)
//         // --env=jsdom
//         // "test": "react-scripts test",
//     })
// })

