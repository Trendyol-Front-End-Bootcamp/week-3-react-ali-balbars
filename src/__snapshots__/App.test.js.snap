// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="homepage"
  >
    <div
      class="Header"
    >
      <h1>
        Rick and Morty
      </h1>
    </div>
    <div>
      <div
        class="Filter"
      >
        <form
          class="form-filter"
        >
          <div
            class="search-container"
          >
            <input
              placeholder="Character Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="dropdowns"
          >
            <div
              class="Dropdown"
            >
              <div
                class="dropdown-btn"
                tabindex="1"
              >
                gender
                <img
                  height="15"
                  src="https://img.icons8.com/ios-glyphs/30/000000/sort-down.png"
                  width="15"
                />
              </div>
              <div
                class="dropdown-content"
              >
                <div
                  class="dropdown-item"
                >
                  All
                </div>
                <div
                  class="dropdown-item"
                >
                  Male
                </div>
                <div
                  class="dropdown-item"
                >
                  Female
                </div>
                <div
                  class="dropdown-item"
                >
                  Unknown
                </div>
              </div>
            </div>
            <div
              class="Dropdown"
            >
              <div
                class="dropdown-btn"
                tabindex="1"
              >
                species
                <img
                  height="15"
                  src="https://img.icons8.com/ios-glyphs/30/000000/sort-down.png"
                  width="15"
                />
              </div>
              <div
                class="dropdown-content"
              >
                <div
                  class="dropdown-item"
                >
                  All
                </div>
                <div
                  class="dropdown-item"
                >
                  Human
                </div>
                <div
                  class="dropdown-item"
                >
                  Alien
                </div>
                <div
                  class="dropdown-item"
                >
                  Unknown
                </div>
              </div>
            </div>
            <div
              class="Dropdown"
            >
              <div
                class="dropdown-btn"
                tabindex="1"
              >
                aliveness
                <img
                  height="15"
                  src="https://img.icons8.com/ios-glyphs/30/000000/sort-down.png"
                  width="15"
                />
              </div>
              <div
                class="dropdown-content"
              >
                <div
                  class="dropdown-item"
                >
                  All
                </div>
                <div
                  class="dropdown-item"
                >
                  Alive
                </div>
                <div
                  class="dropdown-item"
                >
                  Dead
                </div>
                <div
                  class="dropdown-item"
                >
                  Unknown
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        class="CharacterList"
      >
        Loading...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
